hw/core-v-mini-mcu/include/core_v_mini_mcu_pkg.sv.tpl

```diff
@@ -111,14 +107,40 @@ package core_v_mini_mcu_pkg;
       '{ idx: DEBUG_IDX, start_addr: DEBUG_START_ADDRESS, end_addr: DEBUG_END_ADDRESS },
       '{ idx: AO_PERIPHERAL_IDX, start_addr: AO_PERIPHERAL_START_ADDRESS, end_addr: AO_PERIPHERAL_END_ADDRESS },
       '{ idx: PERIPHERAL_IDX, start_addr: PERIPHERAL_START_ADDRESS, end_addr: PERIPHERAL_END_ADDRESS },
-      '{ idx: EXT_SLAVE_IDX, start_addr: EXT_SLAVE_START_ADDRESS, end_addr: EXT_SLAVE_END_ADDRESS },
       '{ idx: FLASH_MEM_IDX, start_addr: FLASH_MEM_START_ADDRESS, end_addr: FLASH_MEM_END_ADDRESS }
   };
 
+  // External slave address map
+  // --------------------------
+  localparam logic [31:0] EXT_SLAVE_START_ADDRESS = 32'h${ext_slave_start_address};
+  localparam logic [31:0] EXT_SLAVE_SIZE = 32'h${ext_slave_size_address};
+  localparam logic [31:0] EXT_SLAVE_END_ADDRESS = EXT_SLAVE_START_ADDRESS + EXT_SLAVE_SIZE;
+
+  // Forward crossbars address map and index
+  // ---------------------------------------
+  // These crossbar connect each muster to the internal crossbar and to the
+  // corresponding external master port.
+  localparam logic [31:0] FWD_XBAR_INT_SLAVE_START_ADDRESS = RAM0_START_ADDRESS;
+  localparam logic [31:0] FWD_XBAR_INT_SLAVE_END_ADDRESS = FLASH_MEM_END_ADDRESS;
+  localparam logic [31:0] FWD_XBAR_INT_SLAVE_SIZE = FWD_XBAR_INT_SLAVE_END_ADDRESS - FWD_XBAR_INT_SLAVE_START_ADDRESS;
+  localparam logic [31:0] FWD_XBAR_INT_SLAVE_IDX = 32'd0;
+
+  localparam logic[31:0] FWD_XBAR_EXT_SLAVE_START_ADDRESS = EXT_SLAVE_START_ADDRESS;
+  localparam logic[31:0] FWD_XBAR_EXT_SLAVE_SIZE = EXT_SLAVE_SIZE;
+  localparam logic[31:0] FWD_XBAR_EXT_SLAVE_END_ADDRESS = EXT_SLAVE_END_ADDRESS;
+  localparam logic[31:0] FWD_XBAR_EXT_SLAVE_IDX = 32'd1;
+
+  // Address map
+  localparam addr_map_rule_t [1:0] FWD_XBAR_ADDR_RULES = '{
+    '{idx: FWD_XBAR_INT_SLAVE_IDX, start_addr: FWD_XBAR_INT_SLAVE_START_ADDRESS, end_addr: FWD_XBAR_INT_SLAVE_END_ADDRESS},
```

Comment:
I would rather add only one rule (external) and the rest assigned by default to the internal bus if possible

Reply:
In [5864cd4](https://github.com/esl-epfl/x-heep/pull/268/commits/5864cd451f935553739fed354c297ab53867f19d) I removed the explicit definition of the internal address rule from `core_v_mini_mcu_pkg`. By default, all the requests on the internal or external address bus that do not meet the external address rule are forwarded to the internal bus (see [here](https://github.com/StMiky/x-heep/blob/5864cd451f935553739fed354c297ab53867f19d/hw/core-v-mini-mcu/system_bus.sv.tpl#L214)). Let me know if that's what you meant.